build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -649,8 +649,7 @@ export const handlers: Handler[] = [
 			const hasScriptsField = Object.prototype.hasOwnProperty.call(json, "scripts");
 
 			if (hasScriptsField) {
-				addPrettier(json);
-				writeFile(file, JSON.stringify(sortPackageJson(json), undefined, 2) + newline);
+				updatePackageJsonFile(path.dirname(file), (json) => addPrettier(json));
```

Comment:
you should really let this function handle opening and pasting the json file, instead of doing it above and in here. Move the above logic into the callback.

Reply:
if you want to have a way for it to optionally skip writing the file, you can compare the input to the output and skip the update in that case (or allow the callback to return something to abort the change).